lib/getstats.js

```diff
@@ -316,6 +323,38 @@ function chromeGetTrackStats(peerConnection, track) {
   });
 }
 
+/**
+ * Get the standardized statistics for a particular MediaStreamTrack in Edge.
+ * @param {RTCPeerConnection} peerConnection
+ * @param {MediaStreamTrack} track
+ * @returns {Promise.<StandardizedTrackStatsReport>}
+ */
+function edgeGetTrackStats(peerConnection, track) {
+  return waitForEdgeIceCompletion(peerConnection).then(function() {
+    return peerConnection.getStats(track);
+  }).then(function(response) {
+    return standardizeEdgeAndFirefoxStats(response);
+  });
+}
+
+function waitForEdgeIceCompletion(peerConnection) {
+  const iceConnectionStatesToWait = new Set(['checking', 'connected', 'new']);
+  if (!iceConnectionStatesToWait.has(peerConnection.iceConnectionState)) {
+    return Promise.resolve();
+  }
+  return new Promise(function(resolve, reject) {
+    const timeoutToFail = setTimeout(reject, 5000, new Error('failed'));
+    peerConnection.addEventListener('iceconnectionstatechange', function onIceConnectionStateChanged() {
+      const iceConnectionStatesToResolve = new Set(['completed', 'failed']);
+      if (iceConnectionStatesToResolve.has(peerConnection.iceConnectionState)) {
+        clearTimeout(timeoutToFail);
+        peerConnection.removeEventListener('iceconnectionstatechange', onIceConnectionStateChanged);
+        setTimeout(resolve, 4000);
```

Comment:
@syerrapragada Can you add a note here as to why you need to resolve after 4 sec?

Reply:
Is 4s necessary, or is a single tick OK? e.g., just

```js
setTimeout(resolve);
```